devito/types/sparse.py

```diff
@@ -89,10 +156,89 @@ def space_order(self):
         """The space order."""
         return self._space_order
 
+    @property
+    def r(self):
+        return self._radius
+
     @property
     def _sparse_dim(self):
         return self.dimensions[self._sparse_position]
 
+    @property
+    def _mpitype(self):
```

Comment:
I didn't know about `MPI._typedict`! We should move the logic this and the property below to a separate tools function, somewhere around this one: https://github.com/devitocodes/devito/blob/master/devito/tools/dtypes_lowering.py#L115

then get the MPI type explicitly by a call to the tool function where needed

Reply:
you can now drop this property

Reply:
up